llm/server.go

```diff
@@ -34,24 +34,35 @@ type LlamaServer struct {
 	done    chan error // Channel to signal when the process exits
 	status  *StatusWriter
 	options api.Options
-}
 
-var cpuOnlyFamilies = []string{
-	"mamba",
+	// TODO - this should be broken down by GPU
+	EstimatedVRAM uint64 // Estimated usage of VRAM by the loaded model
+
+	refMax   int // Maximum concurrent requests
+	refCount int
+	refMu    sync.Mutex
+	refCond  sync.Cond
```

Comment:
[golang.org/x/sync/semaphore](https://pkg.go.dev/golang.org/x/sync/semaphore#) might be useful here